### PR TITLE
postgresql: enable gssapi/kerberos support

### PR DESCRIPTION
**Pull request by georgyo**
###### Motivation for this change

In a kerberized environment, both psql and the server need to be able to support gssapi authentication.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lukegb** [comment]
@ofborg eval

File: pkgs/servers/sql/postgresql/default.nix [hunk 1]
```diff
@@ -8,6 +8,8 @@ let
 
       # This is important to obtain a version of `libpq` that does not depend on systemd.
       , enableSystemd ? (lib.versionAtLeast version "9.6" && !stdenv.isDarwin)
+      , gssSupport ? with stdenv.hostPlatform; !isWindows && !isStatic, libkrb5 ? null
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/default.nix:11]:
```suggestion
      , gssSupport ? with stdenv.hostPlatform; !isWindows && !isStatic, libkrb5
```

libkrb5 should never be null and overriding with null is more of an anti pattern.

**Comment by SuperSandro2000** [comment]
Do we want to target staging?

File: pkgs/servers/sql/postgresql/default.nix (hunk 1 above)
Comment (georgyo) [reply pkgs/servers/sql/postgresql/default.nix]: Rebased on staging, and applied this change.

As a side note, I did copy that method from curl directly:  https://github.com/NixOS/nixpkgs/blob/dc68eb58bb8d4dc3cfb8896932e6146c45579932/pkgs/tools/networking/curl/default.nix#L11

**Comment by marsam** [comment]
@GrahamcOfBorg build postgresql
@GrahamcOfBorg test postgresql

**Review by marsam — APPROVED** [review]

**Pull request merged**